### DISCUSSION TITLE
plotted line is gone after put legend on

### DISCUSSION THREAD

**Discussion by dj-zhou-bg** [alandefreitas/matplotplusplus#172] [General]
I have code like this:
```
struct FigureHandleDj {
    figure_handle f;
    uint8_t       row;
    uint8_t       col;
    uint8_t       p_n;  // plot number
};

FigureHandleDj newFigure(bool t, uint8_t sub_row, uint8_t sub_col) {
    FigureHandleDj fh;
    fh.f   = figure(t);
    fh.row = sub_row;
    fh.col = sub_col;
    fh.p_n = 0;
    return fh;
}

void plotInFigure(FigureHandleDj& fh, uint8_t num, std::vector<double> data[],
                  std::vector<std::string> lgd, uint8_t groups = 1,
                  double x_scale = 0.01) {
    subplot(fh.row, fh.col, num);
    hold(off);
    auto x = linspace(0, x_scale * data[0].size(), data[0].size());
    std::cout << "x.size() = " << x.size() << std::endl;

    for (uint8_t i = 0; i < groups; i++) {
        auto p = plot(x, data[i]);
        hold(on);
        std::cout << "lgd[i].c_str() = " << lgd[i].c_str() << std::endl;
        p->display_name(lgd[i].c_str());
    }
    matplot::legend(lgd);
    fh.f->draw();
}
```
I run the following code to call this function:
```
auto f = newFigure(true, 1, 2);
plotInFigure(f, 0, &data, { "data" });
```
where `data` is a `std::vector<double>` variable. With this setup, I run the program, I only see the legend, but the plotted line from `data` is not shown. However, if I comment out the line ` p->display_name(lgd[i].c_str());` and ` matplot::legend(lgd);`, I can see the line draw from the data, without legend.
How to solve this problem? Thanks

**Comment by denniskb** [comment]
I have the same problem. As soon as I add a legend to my plots, all the contents I plotted up to that point are cleared. Linux + gnuplot.

**Comment by alandefreitas** [comment]
What happens when you run the examples in https://alandefreitas.github.io/matplotplusplus/appearance/labels/legend/ ?

It might be something related to the order of the commands.

**Reply by denniskb** [reply]
The 1st sample works fine :) (didn't try the others). Will play around with `hold()` and `show()` to find out what might be causing it. For now my code does the following:

```c++
figure();
hold(true);
plot(...);
plot(...);
plot(...);
legend(...);
show();
```

I briefly see the plots before they are cleared and the legend is displayed by itself.

**Reply by alandefreitas** [reply]
There might be something in the plot/legend implementation that clears the other.

**Reply by AndrewOrigin** [reply]
in my case, the first example lost the third orange line after legend is displayed. 
![image](https://user-images.githubusercontent.com/55890796/183116797-316c1280-5808-4159-888b-0d40218d116e.png)

**Comment by cameron686** [comment]
I know this is an old thread, but I'm hoping that somebody was able to fix this issues. In my case, I'm plotting several lines. If I don't add a legend, it works perfectly every time. If I add a legend, sometimes it displays correctly, sometimes one of the plot lines disappears, and sometimes all of the plot lines disappear.

In each case, the chart will draw all the plot lines, then when it draws the legend, one or more of the plot lines will disappear. I've tried moving ax->legend before and after "plot," adding ax->draw before and after ax->legend, both interactive and quiet mode, and I still can't get it to work.

I'm using gnuplot 5.4.8 on Windows 10 with MSVC.
